connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -181,6 +181,7 @@ class StreamingQueryProgress private[spark] (
       ("name" -> JString(name)) ~
       ("timestamp" -> JString(timestamp)) ~
       ("batchId" -> JInt(batchId)) ~
+      ("batchDuration" -> JInt(batchDuration)) ~
```

Comment:
Thank you for making a new PR.

However, given the previous discussion on SPARK-44484. We cannot backport a new feature like this under a different `Tests for foreachBatch and Listener` title.

- https://github.com/apache/spark/pull/42077